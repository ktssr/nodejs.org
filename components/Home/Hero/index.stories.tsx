import type { Meta as MetaObj, StoryObj } from '@storybook/react';
import Hero from '.';
import { createNodeReleases } from '@/components/__fixtures__/nodeReleases';

type Story = StoryObj<typeof Hero>;
type Meta = MetaObj<typeof Hero>;

const releases = createNodeReleases();

const currentLTS = releases.find(release => release.isLts);
const currentRelease = releases.find(release => release.status === 'Current');

export const Default: Story = {
  args: {
    title: 'Run JavaScript Everywhere.',
    subTitle:
      'Node.js is a free, open-sourced, cross-platform JavaScript run-time environment that lets developers write command line tools and server-side scripts outside of a browser.',
    ltsVersion: currentLTS?.versionWithPrefix,
    currentVersion: currentRelease?.versionWithPrefix,
  },
};

export default { component: Hero } as Meta;
