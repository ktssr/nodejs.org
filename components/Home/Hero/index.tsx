import { downloadUrlByOS } from '@/util/downloadUrlByOS';
import Link from 'next/link';
import type { FC } from 'react';
import { FormattedMessage } from 'react-intl';
import { useDetectOS } from '@/hooks/useDetectOS';
import styles from './index.module.scss';

type HeroProps = {
  title: string;
  subTitle?: string;
  ltsVersion: string;
  currentVersion: string;
};

const Hero: FC<HeroProps> = ({
  title,
  subTitle,
  ltsVersion,
  currentVersion,
}) => {
  const userOSState = useDetectOS();

  const ltsVersionUrl = downloadUrlByOS(
    ltsVersion,
    userOSState.os,
    userOSState.bitness
  );
  const currentVersionUrl = downloadUrlByOS(
    currentVersion,
    userOSState.os,
    userOSState.bitness
  );

  return (
    <div className={styles.hero}>
      <h1>{title}</h1>
      <h2 className={styles.subTitle}>{subTitle}</h2>
      <div className={styles.buttonsContainer}>
        <div className={styles.downloadLtsContainer}>
          <a className={styles.downloadButton} href={ltsVersionUrl}>
            <FormattedMessage id="components.home.hero.downloadLts" />
            <span>
              <FormattedMessage
                id="components.home.hero.currentVersion"
                values={{ version: ltsVersion }}
              />
            </span>
          </a>
          <p className="t-caption">
            <a href={currentVersionUrl}>
              <FormattedMessage
                id="components.home.hero.getCurrent"
                values={{ version: currentVersion }}
              />
            </a>
          </p>
        </div>
        <Link href="/learn" className={styles.downloadButtonInverse}>
          <FormattedMessage id="components.home.hero.learn" />
        </Link>
      </div>
    </div>
  );
};

export default Hero;
