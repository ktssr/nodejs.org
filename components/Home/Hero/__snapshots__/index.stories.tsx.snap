// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/Hero Default smoke-test 1`] = `
<div>
  <h1>
    Run JavaScript Everywhere.
  </h1>
  <h2>
    Node.js is a free, open-sourced, cross-platform JavaScript run-time environment that lets developers write command line tools and server-side scripts outside of a browser.
  </h2>
  <div>
    <div>
      <a href="https://nodejs.org/dist/v18.16.0/node-v18.16.0.pkg">
        Download Node (LTS)
        <span>
          Version v18.16.0
        </span>
      </a>
      <p>
        <a href="https://nodejs.org/dist/v20.2.0/node-v20.2.0.pkg">
          Download Latest (v20.2.0)
        </a>
      </p>
    </div>
    <a href="/learn">
      Learn Node
    </a>
  </div>
</div>
`;
